src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -420,9 +423,28 @@ bool PolicyHandler::ClearUserConsent() {
   return policy_manager_->ResetUserConsent();
 }
 
-uint32_t PolicyHandler::GetAppIdForSending() const {
+uint32_t PolicyHandler::ChoosePtuApplication(
```

Comment:
What is the purpose of this method if it just returns GetAppIdForSending's response?

Reply:
`GetAppIdForSending` is a const method, so internal variables cannot be changed in the function. This function edits the `last_ptu_app_id_` field so that it can be cached for retry attempts.